### PR TITLE
Prevent saving transformers with duplicate names

### PR DESCRIPTION
**Pull request by P-bibs**
Adds an error message when trying to save a transformer with the same name as an existing transformer

image

### CODE REVIEW

File: src/components/transformer-template/DefinitionCreator.tsx [hunk 1]
```diff
@@ -29,139 +16,56 @@ export default function DefinitionCreator({
...
     } as SavedTransformerContent;
 
-    const savedTransformer = { name, description, content };
+    const savedTransformer = { name, purposeStatement, content };
     const encoded = encodeURIComponent(JSON.stringify(savedTransformer));
 
     const savedUrl = new URL(window.location.toString());
     savedUrl.searchParams.append("transform", encoded);
-
+    console.log(name);
```
Comment (thomascastleman) [inline src/components/transformer-template/DefinitionCreator.tsx:49]: Maybe left by accident?

**Comment by P-bibs** [comment]
Yep this is a good call I'll work on it. One other thing: should we trim all whitespace before saving the transformer's name? We do this to make sure you can't save a transformer with a name that's only whitespace but we don't actually trim whitespace when saving the transformer name.

**Comment by thomascastleman** [comment]
Yeah, I was thinking about that. I think it could be good to both check against and save the trimmed name.

**Comment by P-bibs** [comment]
@thomascastleman added trimming and duplicate suppression for editing saved transformers. Can you take a look when you get the chance?

**Review by thomascastleman — APPROVED** [review]
Looks good now.

One minor annoyance is that if you *edit* a saved Transformer's name to include leading or trailing whitespace, its name does change but if you click edit again the whitespace remains in the text input. It would be nice if the input contents matched the stripped contents, but I don't think this is a huge issue.

**Comment by P-bibs** [comment]
Good catch, I believe that should be fixed now.

**Review by stardust66 — APPROVED** [review]
Looks good!

**Pull request merged**